core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -249,7 +249,14 @@ private[spark] class ExecutorAllocationManager(
    * yarn-client mode when AM re-registers after a failure.
    */
   def reset(): Unit = synchronized {
-    initializing = true
+    /**
+     * When some tasks need to be scheduled and initial executor = 0, resetting the initializing
+     * field may cause it to not be set to false in yarn.
+     * SPARK-20079: https://issues.apache.org/jira/browse/SPARK-20079
+     */
+    if (maxNumExecutorsNeeded() == 0) {
+      initializing = true
```

Comment:
The following code should have a similar function?

```scala
numExecutorsTarget = initialNumExecutors // The default value is 0
numExecutorsToAdd = 1
```
The incoming parameters of the client.requestTotalExecutors method are 1,2,4,8,16...